onnxruntime/core/providers/webgpu/math/cum_sum.cc

```diff
@@ -0,0 +1,94 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include "core/providers/webgpu/math/cum_sum.h"
+#include "core/providers/webgpu/shader_helper.h"
+#include "core/providers/webgpu/webgpu_supported_types.h"
+
+namespace onnxruntime {
+namespace webgpu {
+
+ONNX_OPERATOR_VERSIONED_KERNEL_EX(
+    CumSum,
+    kOnnxDomain,
+    11, 13,
+    kWebGpuExecutionProvider,
+    (*KernelDefBuilder::Create()).TypeConstraint("T", WebGpuSupportedFloatTypes()).TypeConstraint("T2", WebGpuSupportedNumberTypes()).InputMemoryType(OrtMemTypeCPU, 1),
```

Comment:
T2 can only be either int32 or int64.

if you only want to support T2 as int32, use:
```
.TypeConstraint("T2", DataTypeImpl::GetTensorType<int32_t>())
```
otherwise use:
```
.TypeConstraint("T2", {DataTypeImpl::GetTensorType<int32_t>(), DataTypeImpl::GetTensorType<int64_t>()})
```